--- orig\qapplication_win.cpp	Mon Oct 13 21:20:31 1997
+++ qapplication_win.cpp	Mon Oct 13 21:54:27 1997
@@ -115,6 +115,11 @@
 typedef Q_DECLARE(QArrayM,pchar) ArgV;
 
 
+#if defined(QT_TRIAL)
+#include "../../etc/trial/qapplication_trial.cpp"
+#endif // QT_TRIAL
+
+
 /*****************************************************************************
   WinMain() - initializes Windows and calls user's startup function main()
  *****************************************************************************/
@@ -889,12 +894,34 @@
 	}
     }
 
+#if defined(QT_TRIAL)
+    static int random_ofs = 0;
+    msg.message += (random_ofs & 0xffff);
+#endif
+
     if ( msg.message == WM_TIMER ) {		// timer message received
 	do_timers( msg.wParam, &msg );
+#if defined(QT_TRIAL)
+	if ( msg.wParam == (WPARAM)trial_timer ) {
+	    static bool first_time = TRUE;
+	    if ( first_time ) {
+		first_time = FALSE;
+		if ( IsWindowVisible(trial_info->winId()) ) {
+		    delete trial_info;
+		    trial_info = 0;
+		} else {			// punish naughty hacker
+		    random_ofs = QTime::currentTime().secsTo(QTime(0,0,0));
+		}
+	    }
+	}
+#endif
 	return TRUE;
     }
 
-
+#if defined(QT_TRIAL)
+    if ( msg.message == WM_KEYDOWN || msg.message == WM_KEYUP )
+	msg.wParam += trial_messup_1 * 7;
+#endif
     TranslateMessage( &msg );			// translate to WM_CHAR
     DispatchMessage( &msg );			// send to WndProc
     if ( configRequests )			// any pending configs?
@@ -1054,6 +1081,10 @@
 	    RECT     rect;
 	    HBRUSH   brush;
 	    HPALETTE oldPal;
+#if defined(QT_TRIAL)
+	    if ( !trial_messup_2 )
+		break;
+#endif
 	    GetClientRect( hwnd, &rect );
 	    if ( QColor::hPal() ) {
 		oldPal = SelectPalette( hdc, QColor::hPal(), FALSE );
