--- qapplication_win.cpp	Thu Jul 01 15:17:04 1999
+++ qapplication_win.cpp.orig	Thu Jul 01 14:45:20 1999
@@ -186,11 +186,6 @@
 }
 
 
-#if defined(QT_TRIAL)
-#include "qapplication_trial.cpp"
-#endif // QT_TRIAL
-
-
 /*****************************************************************************
   qWinMain() - Initializes Windows. Called from WinMain() in qtmain_win.cpp
  *****************************************************************************/
@@ -1198,34 +1193,11 @@
 	}
     }
 
-#if defined(QT_TRIAL)
-    static int random_ofs = 0;
-    msg.message += (random_ofs & 0xffff);
-#endif
-
     if ( msg.message == WM_TIMER ) {		// timer message received
 	dispatchTimer( msg.wParam, &msg );
-#if defined(QT_TRIAL)
-	if ( msg.wParam == (WPARAM)trial_timer ) {
-	    static bool first_time = TRUE;
-	    if ( first_time ) {
-		first_time = FALSE;
-		if ( IsWindowVisible(trial_info->winId()) ) {
-		    delete trial_info;
-		    trial_info = 0;
-		} else {			// punish naughty hacker
-		    random_ofs = QTime::currentTime().secsTo(QTime(0,0,0));
-		}
-	    }
-	}
-#endif
 	return TRUE;
     }
 
-#if defined(QT_TRIAL)
-    if ( msg.message == WM_KEYDOWN || msg.message == WM_KEYUP )
-	msg.wParam += trial_messup_1 * 7;
-#endif
 
     TranslateMessage( &msg );			// translate to WM_CHAR
     if ( qt_winver == Qt::WV_NT )
@@ -1394,10 +1366,6 @@
 
 	case WM_ERASEBKGND: {			// erase window background
 	    RECT r;
-#if defined(QT_TRIAL)
-	    if ( !trial_messup_2 )
-		break;
-#endif
 	    GetClientRect( hwnd, &r );
 #if defined(QT_ERASE_BACKGROUND)
 	    qt_erase_background
