--- pngrtran.c.orig	2007-09-08 05:22:58.000000000 +0200
+++ pngrtran.c	2007-09-12 00:00:00.000000000 +0200
@@ -549,7 +549,7 @@
    if(png_ptr == NULL) return;
    png_ptr->transformations |= (PNG_EXPAND | PNG_EXPAND_tRNS);
 #ifdef PNG_WARN_UNINITIALIZED_ROW
-   png_ptr->flags &= !(PNG_FLAG_ROW_INIT);
+   png_ptr->flags &= ~(PNG_FLAG_ROW_INIT);
 #endif
 }
 
@@ -578,7 +578,7 @@
    if(png_ptr == NULL) return;
    png_ptr->transformations |= (PNG_EXPAND | PNG_EXPAND_tRNS);
 #ifdef PNG_WARN_UNINITIALIZED_ROW
-   png_ptr->flags &= !(PNG_FLAG_ROW_INIT);
+   png_ptr->flags &= ~(PNG_FLAG_ROW_INIT);
 #endif
 }
 
@@ -591,7 +591,7 @@
    if(png_ptr == NULL) return;
    png_ptr->transformations |= PNG_EXPAND;
 #ifdef PNG_WARN_UNINITIALIZED_ROW
-   png_ptr->flags &= !(PNG_FLAG_ROW_INIT);
+   png_ptr->flags &= ~(PNG_FLAG_ROW_INIT);
 #endif
 }
 #endif
@@ -616,7 +616,7 @@
    png_debug(1, "in png_set_tRNS_to_alpha\n");
    png_ptr->transformations |= (PNG_EXPAND | PNG_EXPAND_tRNS);
 #ifdef PNG_WARN_UNINITIALIZED_ROW
-   png_ptr->flags &= !(PNG_FLAG_ROW_INIT);
+   png_ptr->flags &= ~(PNG_FLAG_ROW_INIT);
 #endif
 }
 #endif /* defined(PNG_READ_EXPAND_SUPPORTED) */
@@ -628,7 +628,7 @@
    png_debug(1, "in png_set_gray_to_rgb\n");
    png_ptr->transformations |= PNG_GRAY_TO_RGB;
 #ifdef PNG_WARN_UNINITIALIZED_ROW
-   png_ptr->flags &= !(PNG_FLAG_ROW_INIT);
+   png_ptr->flags &= ~(PNG_FLAG_ROW_INIT);
 #endif
 }
 #endif
@@ -3867,9 +3867,9 @@
          }
          else if (row_info->bit_depth == 16)
          {
-            png_byte red_high = (trans_value->red > 8) & 0xff;
-            png_byte green_high = (trans_value->green > 8) & 0xff;
-            png_byte blue_high = (trans_value->blue > 8) & 0xff;
+            png_byte red_high = (trans_value->red >> 8) & 0xff;
+            png_byte green_high = (trans_value->green >> 8) & 0xff;
+            png_byte blue_high = (trans_value->blue >> 8) & 0xff;
             png_byte red_low = trans_value->red & 0xff;
             png_byte green_low = trans_value->green & 0xff;
             png_byte blue_low = trans_value->blue & 0xff;
--- pngset.c.orig	2007-09-08 05:22:58.000000000 +0200
+++ pngset.c	2007-09-12 00:00:00.000000000 +0200
@@ -690,7 +690,7 @@
       png_warning(png_ptr, "Insufficient memory to process iCCP chunk.");
       return;
    }
-   png_strncpy(new_iccp_name, name, png_sizeof(new_iccp_name));
+   png_strncpy(new_iccp_name, name, png_strlen(name)+1);
    new_iccp_profile = (png_charp)png_malloc_warn(png_ptr, proflen);
    if (new_iccp_profile == NULL)
    {
@@ -980,7 +980,7 @@
              "Out of memory while processing sPLT chunk");
         }
         /* TODO: use png_malloc_warn */
-        png_strncpy(to->name, from->name, png_strlen(from->name));
+        png_strncpy(to->name, from->name, png_strlen(from->name)+1);
         to->entries = (png_sPLT_entryp)png_malloc_warn(png_ptr,
             from->nentries * png_sizeof(png_sPLT_entry));
         /* TODO: use png_malloc_warn */
